pysolr.py

```diff
@@ -820,7 +829,16 @@ def search(self, q, search_handler=None, **kwargs):
             # cover both cases: there is no response key or value is None
             (decoded.get("response", {}) or {}).get("numFound", 0),
         )
-        return self.results_cls(decoded)
+
+        if "cursorMark" in params and params["cursorMark"] != \
```

Comment:
I'm wondering whether this would be cleaner as `cursorMark = params.get("cursorMark")`, etc. especially with the fallback logic on the `decoded.get()` call.

Reply:
Maybe this is better?:
```
        cursorMark = params.get("cursorMark", None)
        if cursorMark != decoded.get("nextCursorMark", cursorMark):
```

Reply:
I like that and it avoids an extra lookup

Reply:
cool